### ISSUE TITLE
Carriage return characters break generated C# code

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi#13886]
Ali Cloud docs have an example with the following snippet:

```
# add index for logstore, which is used to query logs
locals {
  sls_default_token = ", '\";=()[]{}?@&<>/:\n\t\r"
}
```

This is translated to the following C# code in our program gen:

```
    /// 
    ///     var slsDefaultToken = @", '"";=()[]{}?@&amp;&lt;&gt;/:
    /// 	
";
    /// 
```

which fails to compile.

This is blocking the Ali Cloud upgrade.

**Comment by Zaid-Ajaj** [comment]
Converting the C# program seems work file using latest `pulumi convert`:
```tf
locals {
  sls_default_token = ", '\";=()[]{}?@&<>/:\n\t\r"
}
```
Then
```
pulumi convert --from terraform --language csharp --out csharp
```
Gives:
```csharp
using System.Collections.Generic;
using System.Linq;
using Pulumi;

return await Deployment.RunAsync(() => 
{
    var slsDefaultToken = @", '"";=()[]{}?@&<>/:
	
";

});
```
And it compiles fine when building locally. 

@mikhailshilkov Are we using the new tf converter or the old tf2pulumi here?

**Comment by Frassle** [comment]
All examples are still using the old converter. I've spoken to @t0yv0 about a plan to switch over, the first part of which is unblocked by https://github.com/pulumi/pulumi-converter-terraform/pull/29. Anton was going to try replacing the convert code in tfgen with a write out of all the examples to one json file then a call to `pulumi convert --from terraform --language pcl -- --convert-examples=examples.json` to get them all converted to pcl at once, and then running their normal per language generators on that (Fixing that will be a later phase of work).

N.B. `--language pcl` because that means convert just copies the files from the converter plugin and doesn't try to do anything with them because the only file written out by the converter for this call will be some examples,json with all the PCL in it instead.

**Comment by t0yv0** [comment]
Tracked in https://github.com/pulumi/pulumi-terraform-bridge/issues/1280 - still the plan to do it as soon as I free up.

**Comment by Zaid-Ajaj** [comment]
Until we get the examples to convert using the new converter, does the customer need to skip generating this particular example for C# in the meantime?

**Comment by justinvp** [comment]
I'm going to close this as no repro, since it's already fixed in the new converter.

**Comment by guineveresaenger** [comment]
This is still a bug. 

I came across this and the assertion that this was fixed for Alicloud. Generating examples with carriage returns generates SDK examples that are not properly commented out as examples:

```
  /// ```csharp
    /// using System.Collections.Generic;
    /// using System.Linq;
    /// using Pulumi;
    /// using AliCloud = Pulumi.AliCloud;
    /// using Random = Pulumi.Random;
    /// 
    /// return await Deployment.RunAsync(() =&gt; 
    /// {
    ///     var @default = new Random.Index.Integer("default", new()
    ///     {
    ///         Max = 99999,
    ///         Min = 10000,
    ///     });
    /// 
    ///     var defaultProject = new AliCloud.Log.Project("default", new()
    ///     {
    ///         ProjectName = $"example-value-{@default.Result}",
    ///     });
    /// 
    ///     var defaultStore = new AliCloud.Log.Store("default", new()
    ///     {
    ///         ProjectName = defaultProject.ProjectName,
    ///         LogstoreName = "example-value",
    ///     });
    /// 
    ///     var slsDefaultToken = @", '"";=()[]{}?@&amp;&lt;&gt;/:
    /// 	
";
    /// 
    ///     var example = new AliCloud.Log.StoreIndex("example", new()
    ///     {
    ...
```

**Comment by justinvp** [comment]
Ah, it looks like the problem isn't the example code itself, e.g. this is valid C#:

```c#
var slsDefaultToken = @", '"";=()[]{}?@&<>/:
	
";
```

The `@` verbatim string allows newlines.

It looks like the problem is when the example code is put into a `///` doc comment.

If we were handling the newline within the comment correctly, it'd be this:

```
    ///     var slsDefaultToken = @", '"";=()[]{}?@&amp;&lt;&gt;/:
    /// 	
    /// ";
```

But the `\n\t\r` might get "lost" if/when someone copies/pastes the example, so it's probably best to fix this so it's emitted as:

```
    ///     var slsDefaultToken = ", '"";=()[]{}?@&amp;&lt;&gt;/:\n\t\r";
```
